ontology/coap.ttl

```diff
@@ -58,7 +58,7 @@ cov:hopLimit a rdf:Property, owl:DatatypeProperty ;
 
 cov:contentFormat a rdf:Property, owl:DatatypeProperty ;
                   rdfs:range xsd:unsignedShort ;
-                  rdfs:comment "Indicates a numeric CoAP Content-Format [[RFC7252]] as a combination of Media Type and Content Coding. Must match the given <code>contentType</code> and <code>contentCoding</code> members."@en ;
+                  rdfs:comment "Indicates a numeric CoAP Content-Format [[RFC7252]] as a combination of Media Type and Content Coding as a guideline for Consumers without knowledge of the values registered with IANA. Must match the given <code>contentType</code> and <code>contentCoding</code> members."@en ;
```

Comment:
Would you consider this an improvement? We could probably also add a reference to [appendix C](https://w3c.github.io/wot-thing-description/#contentType-usage) of the TD specification somewhere.

```suggestion
                  rdfs:comment "Indicates a numeric CoAP Content-Format [[RFC7252]] as a combination of Media Type and Content Coding as a guideline for Consumers without knowledge of the values registered with IANA. Based on this information, Consumers can choose an appropriate Form for an Interaction Affordance and then use the correct Content-Format in CoAP requests. Must match the given <code>contentType</code> and <code>contentCoding</code> members."@en ;
```

Reply:
That's a step in the right direction, I think. However, I'm not sure if a client would actually use `cov:contentFormat` to select the most appropriate form, or if the selection was made via `contentType` and the client then just used `cov:contentFormat` to construct the request. Perhaps a good next step would be to first clarify the purpose of `contentType` in a form. Is it really just for choosing the form that gives the client the best response, or does it also affect how the client should construct its request?

Reply:
> Is it really just for choosing the form that gives the client the best response, or does it also affect how the client should construct its request?

It would also prescribe which content type/coding/format would be used for the request body/payload, so if a `contentType` of `application/cbor` was included in a form for an action that defines an `input` data schema, then the client would need to use CBOR for encoding the data that is required for invoking that action. I tried to express that in the second half of the suggestion above, but there is probably room for further clarification.

Reply:
```suggestion
                  rdfs:comment "Prescribes which content type/coding/format is to be used for the request body/payload"@en ;
```

Reply:
Hmm, I think that this would actually be an oversimplification since the purpose of contentType can vary based on the presence of, for instance, `input`, `output`, and `response` (also see the TD appendix cited above). I think you could argue that Things are a bit complicated here.

But maybe something like this could work? (I kept the relationship with `contentType` and `contentCoding` here.)

```suggestion
                  rdfs:comment "Prescribes the Content-Format to be used with CoAP request or response payloads. Must match the given <code>contentType</code> and <code>contentCoding</code> members in the same form."@en ;
```

Reply:
Yes, I think that's getting close! Thanks for putting so much thought into it. And I agree that it's difficult to capture the purpose of `contentType` in a concise sentence. But if we can't even explain it in a sentence, how can we expect readers to understand it? 😊 

Maybe like this?

```suggestion
                  rdfs:comment "Specifies the content format to be used by a client for input and the content format to be used by a server for output (where applicable). The value must be the numeric Content-Format ID registered with IANA for the combination of the <code>contentType</code> and <code>contentCoding</code> specified in the same form."@en ;
```

Reply:
That looks very good! :) Thank you!

Reply:
I really like this last solution, thanks for the cooperation @ektrah and @JKRhb !!